src/main/java/edu/ie3/datamodel/models/timeseries/repetitive/RepetitiveTimeSeries.java

```diff
@@ -34,6 +30,22 @@ protected RepetitiveTimeSeries(UUID uuid, Set<E> entries) {
 
   @Override
   public Optional<V> getValue(ZonedDateTime time) {
-    return Optional.of(calc(time));
+    return Optional.ofNullable(calc(time));
+  }
+
+  @Override
+  protected Optional<ZonedDateTime> getPreviousDateTime(ZonedDateTime time) {
+    return Optional.of(time.minusHours(1));
```

Comment:
This is only working for hourly time series, or?

Reply:
Since this is an abstract class, I will remove this implementation. It should instead be implemented by a sub-class.